src/serialize.rs

```diff
@@ -0,0 +1,103 @@
+        .bind(
+            ACTION_AWESOME_SUPER_SELECT,
+            vec![KeyCode::LAlt, KeyCode::LControl, KeyCode::Return],
+        );
+        map.save_to_path(TESTPATH).expect("Failed to save string InputMap");
+        let new_map = InputMap::<String>::new_from_path(TESTPATH).expect("Failed to load string InputMap");
+        assert_eq!(map.actions, new_map.actions);
+        map.load_from_path(TESTPATH).expect("Failed to Load from path");
+        assert_eq!(map.actions, new_map.actions);
+    }
+
+    fn test_inputmap_enum(){
+        let mut map = super::InputMap::<TestAction>::default();
+        map.bind(TestAction::Select, KeyCode::Return)
+        .bind(TestAction::Select, GamepadButtonType::South)
+        .bind(TestAction::SuperSelect, vec![KeyCode::LAlt, KeyCode::Return])
+        .bind(TestAction::SuperSelect, vec![KeyCode::RAlt, KeyCode::Return])
+        // This should bind left/right control and left/right alt, but the combos would get ridiculous so hopefully this is sufficient.
+        .bind(
+            TestAction::AwesomeSuperSelect,
+            vec![KeyCode::LAlt, KeyCode::LControl, KeyCode::Return],
+        );
+        map.save_to_path(TESTPATH).expect("Failed to save enum InputMap");
+        let new_map = InputMap::<TestAction>::new_from_path(TESTPATH).expect("Failed to load enum InputMap");
+        assert_eq!(map.actions, new_map.actions);
+        map.load_from_path(TESTPATH).expect("Failed to Load from path");
+        assert_eq!(map.actions, new_map.actions);
+    }
+    
+    #[test]
+    fn it_works(){
+        test_binding(KeyCode::Space);
+        test_binding(vec![KeyCode::Space, KeyCode::LControl]);
+        test_binding(GamepadButtonType::North);
+        test_binding(vec![GamepadButtonType::North, GamepadButtonType::LeftThumb]);
+        test_action(Action{
+            bindings : vec![KeyCode::Space.into(), KeyCode::J.into()]
+        });
+        test_inputmap_string();
+        test_inputmap_enum();
+        std::fs::remove_file(TESTPATH).expect("Failed to remove testsave file");
+    }
+}
+
+impl<T : Eq + Hash> InputMap<T> {
+    pub fn save_to_path(&self, path : &str) -> std::io::Result<()> where T : Serialize{
+        let contents = ron::ser::to_string_pretty(&self.actions, ron::ser::PrettyConfig::default()).expect("There was an error making the string");
+        std::fs::write(path, contents)
+    }
+    pub fn load_from_path(&mut self, path : &str) -> std::io::Result<()> where T : DeserializeOwned{
+        let ron_string = std::fs::read_to_string(path)?;
+        let actions = ron::from_str(&ron_string).expect("Failed to get actions from ron string");
+        self.actions = actions;
+        //may need to clear self here but i dont really know what that does
+        Ok(())
+    }
+    pub fn new_from_path(path : &str) -> std::io::Result<InputMap<T>> where T : DeserializeOwned{
```

Comment:
Would it make sense to use actual `Path`s here?

Reply:
It should be `new_from_path<T: Into<Path>>(path T)` but I don't know how to do that, because when I tried it told me Path is a private type; maybe I was just getting the wrong Path Type.